cluster/v1beta1/helpers.go

```diff
@@ -82,3 +83,85 @@ func BuildClusterSelector(clusterSet *ManagedClusterSet) (labels.Selector, error
 		return nil, fmt.Errorf("selectorType is not right: %s", clusterSet.Spec.ClusterSelector.SelectorType)
 	}
 }
+
+type PlacementDecisionGetter interface {
+	List(selector labels.Selector, namespace string) (ret []*PlacementDecision, err error)
+}
+
+type PlacementDecisionClustersTracker struct {
+	placement               *Placement
+	placementDecisionGetter PlacementDecisionGetter
+	decisionClusters        sets.String
+}
+
+type PlacementDecisionClustersResponse struct {
+	addeddDecisionClusters  sets.String
+	deletedDecisionClusters sets.String
+}
+
+func (pdcr *PlacementDecisionClustersResponse) Added() sets.String {
+	return pdcr.addeddDecisionClusters
+}
+
+func (pdcr *PlacementDecisionClustersResponse) Deleted() sets.String {
+	return pdcr.deletedDecisionClusters
+}
+
+func NewPlacementDecisionClustersTracker(placement *Placement, pdl PlacementDecisionGetter) (*PlacementDecisionClustersTracker, error) {
+	return &PlacementDecisionClustersTracker{
+		placement:               placement,
+		placementDecisionGetter: pdl,
+	}, nil
+}
+
+// get() return the placement decision cluster names, and do not persist it in tracker's decisionClusters
+func (pdct *PlacementDecisionClustersTracker) get() (sets.String, error) {
```

Comment:
I think you could have a Get()(sets.string, sets.sting, error) that returns added and removed. In the func, you would need a lock to avoid racy; get the update-to-date decision, compared with and update existing and return.

Refresh/Added/Deleted may not be needed.

Reply:
Removed Refresh/Added/Deleted and add rw lock to avoid racy.